doc/en/docs/autograd_doc.md

```diff
@@ -0,0 +1,232 @@
+﻿# singa.autograd
+
+This part will present an overview of how autograd works and give a simple example of neuron network which implemented by using autograd API. 
+## Autograd Mechanics
+To get clear about how autograd system works, we should understand three important concepts in this system, they are `singa.tensor.Tensor` , `singa.autograd.Operation`, and `singa.autograd.Layer`.  For briefness, these three classes will be denoted as `tensor`, `operation`, and `layer`.
+### Tensor
+The class `tensor` has three attributes which are important in autograd system, they are `.creator`, `.requires_grad`, and `.stores_grad`.
+-  `tensor.creator` should be an `operation` object. It records the particular `operation` which generates the `tensor ` itself.
+-  `.requires_grad` and `.stores_grad` are both boolean indicator. These two attributes record whether a `tensor` needs gradients and whether gradients of a  `tensor` need to be stored when do backpropagation. It should be noted that if `.stores_grad` is true, then `.requires_grad` must be true， not vice versa.
+### Operation
+`operation` is the fundamental element of computation graph. A `operation` takes one or more `tensor` as input, and then output one or more `tensor`. The class `operation` has four important member functions:
+ 1.  `._do_forward()` Inputs and outputs of this function are both  `tensor`. This function will be invoked when the `operation` is called. So this function can be considered being controlled indirectly by users when they define their networks. This function mainly do two things: 
```

Comment:
no need to expose the _do_forward() function to users. remove the documentation.

Reply:
the same for _do_backward()